_search-plugins/search-pipelines/rag-processor.md

```diff
@@ -0,0 +1,97 @@
+`tag` | String | The processor's identifier. Optional.
+`description` | String | A description of the processor. Optional.
+
+### Context field list
+
+The `context_field_list` is a list of fields in document sources that the pipeline uses as context for RAG. For example, let's say your OpenSearch index stores a collection of documents, each containing a `title` and `text`:
+
+```json
+{
+  "_index": "qa_demo",
+  "_id": "SimKcIoBOVKVCYpk1IL-",
+  "_source": {
+    "title": "Abraham Lincoln 2",
+    "text": "Abraham Lincoln was born on February 12, 1809, the second child of Thomas Lincoln and Nancy Hanks Lincoln, in a log cabin on Sinking Spring Farm near Hodgenville, Kentucky.[2] He was a descendant of Samuel Lincoln, an Englishman who migrated from Hingham, Norfolk, to its namesake, Hingham, Massachusetts, in 1638. The family then migrated west, passing through New Jersey, Pennsylvania, and Virginia.[3] Lincoln was also a descendant of the Harrison family of Virginia; his paternal grandfather and namesake, Captain Abraham Lincoln and wife Bathsheba (née Herring) moved the family from Virginia to Jefferson County, Kentucky.[b] The captain was killed in an Indian raid in 1786.[5] His children, including eight-year-old Thomas, Abraham's father, witnessed the attack.[6][c] Thomas then worked at odd jobs in Kentucky and Tennessee before the family settled in Hardin County, Kentucky, in the early 1800s.[6]\n"
+  }
+}
+```
+
+You can specify to send only the `text` contents to the LLM by setting `"context_field_list": ["text"]` in the processor. 
+
+## Example 
+
+The following example demonstrates using a search pipeline with a `retrieval_augmented_generation` processor. 
+
+### Creating a search pipeline 
+
+The following request creates a search pipeline containing a `retrieval_augmented_generation` processor for an OpenAI model:
+
+```json
+PUT /_search/pipeline/rag_pipeline
+{
+  "response_processors": [
+    {
+      "retrieval_augmented_generation": {
+        "tag": "openai_pipeline_demo",
+        "description": "Demo pipeline Using OpenAI Connector",
+        "model_id": "gnDIbI0BfUsSoeNT_jAw",
+        "context_field_list": ["text"],
+        "system_prompt": "You are a helpful assistant",
+        "user_instructions": "Generate a concise and informative answer in less than 100 words for the given question"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Using a search pipeline
+
+Combine an OpenSearch query with an `ext` object that holds generative question answering parameters for the LLM:
+
+```json
+GET /my_rag_test_data/_search?search_pipeline=rag_pipeline
+{
+  "query": {
+    "match": {
+      "text": "Abraham Lincoln"
+    }
+  },
+  "ext": {
+    "generative_qa_parameters": {
+      "llm_model": "gpt-3.5-turbo",
+      "llm_question": "Was Abraham Lincoln a good politician",
+      "conversation_id": "iXC4bI0BfUsSoeNTjS30",
+      "context_size": 5,
+      "interaction_size": 5,
```

Comment:
Same thing, can you update "interaction_size" to "message_size"? I remember this rename is suggested by you.